inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -71,7 +74,15 @@ def arxiv_fulltext_download(obj, eng):
     )
 
     if pdf:
-        obj.log.info('PDF retrieved from arXiv for %s', arxiv_id)
+        if pdf.get_version().mimetype == 'application/pdf':
+            obj.log.info('PDF retrieved from arXiv for %s', arxiv_id)
+        else:
+            # We need to delete the failed PDF
+            del obj.files[filename]
+            pdf.delete()
```

Comment:
I'd say better safe then sorry. My head was exploding in trying to understand the what would happen in plainly overriding. Doing both is anyway not wrong.